include/s3select_functions.h

```diff
@@ -1777,23 +1776,26 @@ bool operator()(bs_stmt_vec_t* args, variable* result) override
         throw base_s3select_exception("esacpe expression must be string");
       }
 
-      std::vector<char> like_as_regex = transform(like_expr_val.str(), *escape_expr_val.str());
-      std::string like_as_regex_str(like_as_regex.begin(), like_as_regex.end());
-      compiled_regex = std::regex(like_as_regex_str);
+      like_as_regex = transform(like_expr_val.str(), *escape_expr_val.str());
     }
 
     value main_expr_val = main_expr->eval();
     if (main_expr_val.type != value::value_En_t::STRING)  {
         throw base_s3select_exception("main expression must be string");
     }
-         
-    if ( std::regex_match(main_expr_val.to_string(), compiled_regex)) {
+
+    std::string content = main_expr_val.to_string();
+    std::string like_as_regex_str(like_as_regex.begin(), like_as_regex.end());
+    RE2 compiled_regex(like_as_regex_str);
```

Comment:
`compiled_regex` should be a class member 
upon constant_state is true it `compiled_regex` is compiled once instead of many times.